src/commands/publish/mod.rs

```diff
@@ -17,18 +18,30 @@ use crate::commands::kv;
 use crate::commands::subdomain::Subdomain;
 use crate::http;
 use crate::settings::global_user::GlobalUser;
+
 use crate::settings::target::Target;
 use crate::terminal::{emoji, message};
 
 pub fn publish(
     user: &GlobalUser,
-    target: &Target,
+    target: &mut Target,
```

Comment:
we need to ask the target to update its in-memory kv_namespaces in the case that sites is specified. we do this here because we want to delay this logic to preview (implementing later) and publish, rather than requiring it for every command.

Reply:
do we write this change back out to wrangler.toml?

Reply:
no.

Reply:
cool, this makes sense!